### PR TITLE
[JENKINS-43930 & Others] - FindBugs fixes towards 1.9

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Update Parent POM and increase the FindBugs effort
- [x] Fix file descriptor leak, which may require agent JVM restart to retry the installation. [JENKINS-43930](https://issues.jenkins-ci.org/browse/JENKINS-43930)
- [x] Refactor Restarter thread and improve logging. Makes FindBucs happy about non-static inner classes
- [x] Update changelog

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java [hunk 1]
```diff
@@ -283,6 +273,32 @@ public static String generateSlaveXml(String id, String java, String vmargs, Str
...
+
+        @Override
+        public void run() {
+            try {
+                StreamTaskListener task = StreamTaskListener.fromStdout();
+                int r = runElevated(agentExe, "start", task, rootDir);
+                task.getLogger().println(r == 0 ? "Successfully started" : "Start service failed. Exit code=" + r);
+            } catch (IOException | InterruptedException ex) {
+                // Level is severe, because the process won't be recovered in the service mode
+                LOGGER.log(Level.SEVERE, "Failed to start the service with id=" + serviceId, ex);
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java]: The only change is logging here

File: pom.xml [hunk 2]
```diff
@@ -66,6 +68,14 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <configuration>
```
Comment (jglick) [inline pom.xml]: Why?

File: src/main/java/org/jenkinsci/modules/windows_slave_installer/SlaveInstallerFactoryImpl.java [hunk 3]
```diff
@@ -26,7 +27,9 @@ public SlaveInstaller createIfApplicable(Channel c) throws IOException, Interrup
         return null;
     }
 
+    @SuppressFBWarnings(value = "SE_NO_SERIALVERSIONID", justification = "Remoting does not need it")
```
Comment (jglick) [inline src/main/java/org/jenkinsci/modules/windows_slave_installer/SlaveInstallerFactoryImpl.java]: Easier to just add it, no?

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: The default threshold in plugin-pom is `Medium`, hence it misses the File leak on exceptional path in this case. Ideally should be fixed in Parent POM, of course

File: src/main/java/org/jenkinsci/modules/windows_slave_installer/SlaveInstallerFactoryImpl.java (hunk 3 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/modules/windows_slave_installer/SlaveInstallerFactoryImpl.java]: Maybe it is safe in this case, not sure

File: pom.xml (hunk 2 above)
Comment (jglick) [inline pom.xml]: File leaks are treated as low severity by FindBugs? Maybe that is what should be fixed.

Comment (oleg-nenashev) [inline pom.xml]: * File leaks on the common path are the Medium/High Severity depending on the detector
* File leaks on Exceptional paths are low

The latter bullet does not make me happy as well

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Pull request merged**